openedx/core/djangoapps/programs/tests/test_utils.py

```diff
@@ -811,3 +812,26 @@ def test_can_enroll(self, can_enroll, mock_has_access):
         data = ProgramMarketingDataExtender(self.program, self.user).extend()
 
         self.assertEqual(data['courses'][0]['course_runs'][0]['can_enroll'], can_enroll)
+
+    def test_learner_eligibility_for_one_click_purchase(self):
+        """
+        Learner should be eligible for one click purchase if:
+        - program is eligible for one click purchase
+        - learner is not enrolled in any of the course runs associated with the program
+        """
+        data = ProgramMarketingDataExtender(self.program, self.user).extend()
+        self.assertTrue(data['is_learner_eligible_for_one_click_purchase'])
+
+        courses = [self._create_course(self.course_price)]
+
+        program = ProgramFactory(
+            courses=courses,
+            is_program_eligible_for_one_click_purchase=False
+        )
+        data = ProgramMarketingDataExtender(program, self.user).extend()
+        self.assertFalse(data['is_learner_eligible_for_one_click_purchase'])
+
+        courses.append(self._create_course(self.course_price, is_enrolled=True))
+        program2 = ProgramFactory(courses=courses)
```

Comment:
Explicitly set `is_program_eligible_for_one_click_purchase=True` here to avoid future confusion and jumping between files.